### PR TITLE
migrate spring cloud starter repo

### PR DESCRIPTION
**Pull request by moarychan**
Work Item:
https://dev.azure.com/azure-spring-integration/spring-integration-private/_workitems/edit/263

### CODE REVIEW

File: eng/versioning/version_client.txt [hunk 1]
```diff
@@ -100,6 +100,13 @@ com.microsoft.azure:spring-cloud-azure-feature-management;1.2.8-beta.1;1.2.8-bet
 com.microsoft.azure:spring-cloud-azure-keyvault-config;1.2.8-beta.1;1.2.8-beta.1
 com.microsoft.azure:spring-cloud-azure-messaging;1.2.8-beta.1;1.2.8-beta.1
 com.microsoft.azure:spring-cloud-azure-storage;1.2.8-beta.1;1.2.8-beta.1
+com.microsoft.azure:spring-cloud-starter-azure-appconfiguration-config;1.2.8-beta.1;1.2.8-beta.1
+com.microsoft.azure:spring-cloud-starter-azure-eventhubs-kafka;1.2.8-beta.1;1.2.8-beta.1
+com.microsoft.azure:spring-cloud-starter-azure-eventhubs;1.2.8-beta.1;1.2.8-beta.1
+com.microsoft.azure:spring-cloud-starter-azure-servicebus;1.2.8-beta.1;1.2.8-beta.1
+com.microsoft.azure:spring-cloud-starter-azure-storage-queue;1.2.8-beta.1;1.2.8-beta.1
+com.microsoft.azure:spring-starter-azure-cache;1.2.8-beta.1;1.2.8-beta.1
+com.microsoft.azure:spring-starter-azure-storage;1.2.8-beta.1;1.2.8-beta.1
```
Comment (JimSuplizio) [inline eng/versioning/version_client.txt:109]: @saragluna these artifacts need to get added to the work item for renaming.

File: sdk/spring/azure-spring-cloud-starter-eventhubs-kafka/pom.xml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    <version>1.7.0</version> <!-- {x-version-update;com.azure:azure-client-sdk-parent;current} -->
+    <relativePath>../../parents/azure-client-sdk-parent</relativePath>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.microsoft.azure</groupId>
+  <artifactId>spring-cloud-starter-azure-eventhubs-kafka</artifactId>
+  <version>1.2.8-beta.1</version> <!-- {x-version-update;com.microsoft.azure:spring-cloud-starter-azure-eventhubs-kafka;current} -->
+
+  <name>Spring Cloud Azure Storage</name>
```
Comment (JimSuplizio) [inline sdk/spring/azure-spring-cloud-starter-eventhubs-kafka/pom.xml:17]: nit: This should be Event Hubs Kafka not Storage.

File: sdk/spring/azure-spring-cloud-starter-eventhubs/pom.xml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    <version>1.7.0</version> <!-- {x-version-update;com.azure:azure-client-sdk-parent;current} -->
+    <relativePath>../../parents/azure-client-sdk-parent</relativePath>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.microsoft.azure</groupId>
+  <artifactId>spring-cloud-starter-azure-eventhubs</artifactId>
+  <version>1.2.8-beta.1</version> <!-- {x-version-update;com.microsoft.azure:spring-cloud-starter-azure-eventhubs;current} -->
+
+  <name>Azure Spring Cloud Starter Event Hubs</name>
```
Comment (JimSuplizio) [inline sdk/spring/azure-spring-cloud-starter-eventhubs/pom.xml:17]: Should all of these starters have "Starter" in their names? There's an inconsistency in the <name> strings with the libraries in this PR.

File: sdk/spring/azure-spring-cloud-starter-storage/pom.xml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    <version>1.7.0</version> <!-- {x-version-update;com.azure:azure-client-sdk-parent;current} -->
+    <relativePath>../../parents/azure-client-sdk-parent</relativePath>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.microsoft.azure</groupId>
+  <artifactId>spring-starter-azure-storage</artifactId>
+  <version>1.2.8-beta.1</version> <!-- {x-version-update;com.microsoft.azure:spring-starter-azure-storage;current} -->
+
+  <name>Azure Spring Cloud Appconfiguration Config</name>
```
Comment (JimSuplizio) [inline sdk/spring/azure-spring-cloud-starter-storage/pom.xml:17]: this looks like a copy/paste from Azure Spring Cloud Appconfiguration Config

File: eng/versioning/version_client.txt (hunk 1 above)
Comment (chenrujun) [reply eng/versioning/version_client.txt:109]: I added in this issue: https://github.com/Azure/azure-sdk-for-java/issues/13523

**Comment by saragluna** [comment]
@JimSuplizio I will merge this PR now. We'll file another issue to change all our `<name>` tags to make sure they are correct and consistent.

File: sdk/spring/azure-spring-cloud-starter-eventhubs/pom.xml (hunk 3 above)
Comment (chenrujun) [reply sdk/spring/azure-spring-cloud-starter-eventhubs/pom.xml:17]: Thank you for your careful check.
We plan to update name after artifact name updated.
Here is new created issue: https://github.com/Azure/azure-sdk-for-java/issues/13953

**Pull request merged**